includes/classes/Category.php

```diff
@@ -0,0 +1,168 @@
+<?php
+declare(strict_types=1);
+
+/**
+ * @copyright Copyright 2003-2024 Zen Cart Development Team
+ * @license http://www.zen-cart.com/license/2_0.txt GNU Public License V2.0
+ * @version $Id: DrByte 2024 Jul 23 New in v2.1.0-alpha1 $
+ * based on Product
+ * @var language $lng
+ * @var queryFactory $db
+ */
+
+use Zencart\Traits\NotifierManager;
+
+class Category
+{
+    use NotifierManager;
+
+    protected array $data;
+    protected array $languages;
+
+    /** @deprecated use ->get('property') or ->getData()  */
+    public array $fields;
+
+    /** @deprecated use !exists()  */
+    public bool $EOF = true;
+
+    public function __construct(protected ?int $categories_id = null)
+    {
+        $this->initLanguages();
+
+        if ($this->categories_id !== null) {
+            $this->data = $this->loadProductDetails($this->categories_id);
```

Comment:
Shouldn't this method be named `loadCategoryDetails`?  There's nothing product-specific in the data returned.